R/hub-connection.R

```diff
@@ -0,0 +1,178 @@
+#' Connect to a Modeling Hub
+#'
+#'
+#' @param hub_path Path to the Modeling Hub directory.
+#' @param hubmeta_path Path to a hubmeta file. The default (`NULL`) expects a file
+#'   in the root of the directory given in `hub_path` named `hubmeta` and having
+#'   an accepted file format (see `hubmeta_format`). Use this argument if the
+#'   hubmeta file is in a non-default location or has a non-default file name and
+#'   include the file extension in the path.
+#' @param hubmeta_format The file extension of the hubmeta file. One of `"json"`,
+#'   `"yaml"` or `"yml"`. Defaults to `"json"` and ignored when `hubmeta_path` is
+#'   provided.
+#' @return an S3 object of class `<hub-connection>` containing metadata about the
+#'   modeling hub. This object is used to communicate with the Modelling Hub when
+#'   extracting or submitting data.
+#' @export
+#'
+#' @examples
+#' # Simple forecasting Hub example
+#' hub_path <- system.file("hub_1", package = "hubUtils")
+#' connect_hub(hub_path)
+#' # Scenario Hub example
+#' scnr_path <- system.file("scnr_hub_1", package = "hubUtils")
+#' connect_hub(hub_path)
+connect_hub <- function(hub_path, hubmeta_path = NULL,
+                        hubmeta_format = c("json", "yaml", "yml")) {
+
+    hubmeta_format <- rlang::arg_match(hubmeta_format)
+
+    if (is.null(hubmeta_path)) {
+        hubmeta_path <- fs::path(
+            hub_path,
+            "hubmeta",
+            ext = hubmeta_format
+            )
+    }
+
+    hubmeta <- read_hubmeta(hubmeta_path)
+
+    hubmeta[["hubmeta_path"]] <- hubmeta_path
+    hubmeta[["hub_path"]] <- hub_path
+
+    new_hub_connection(hubmeta)
+
+}
+
+# INTERNAL ----
+
+#' Create new hub connection S3 class object
+#'
+#' Internal function for create a new hub connection S3 class object
+#' @param x A metadata list generated by [read_hubmeta()].
+#'
+#' @return A hub connection S3 class object.
+#' @noRd
+new_hub_connection <- function(x) {
+    stopifnot(is.list(x))
+
+    x <- assing_hc_attrs(x)
```

Comment:
```suggestion
    x <- assign_hc_attrs(x)
```

Reply:
Doh! well spotted typo! Corrected in 44ac1ac39cfbf6f2317bfe2d78d6940558809abd